internal/overlord/servstate/handlers.go

```diff
@@ -162,7 +168,20 @@ func (m *ServiceManager) doStart(task *state.Task, tomb *tomb.Tomb) error {
 			}
 			releasePlan()
 		}
-		return fmt.Errorf("cannot start service: exited quickly with code %d", cmd.ProcessState.ExitCode())
+
+		it := logBuffer.HeadIterator(lastLogLines)
```

Comment:
That's confused me several times as well, because usually we think of "tail" as tailing the end of the log as in the `tail` command. But in the ring buffer the tail is the first logs that were written (it follows the head) and the head is the most recent. Interestingly, I just typed "ring buffer head and tail" into Google and the first hit was a [thread](https://groups.google.com/g/comp.os.msdos.programmer/c/H75qEeKMGVo) with people debating which should be named which!

Reply:
You could avoid calling it head and tail and call it "oldest" and "newest". Might be clearer.

Reply:
Probably clearer, yeah. However, I'm not going to make that change in this PR. @hpidcock can open a separate PR if he wants, though IMO it's not worth it.

Reply:
Heh, I think of this as if it's a lizard "if you cut the head off, then it's dead, but you can cut off the tail as much as you want, it grows back". Anyway, happy to change the naming to be less ambiguous.